samza-core/src/test/scala/org/apache/samza/util/TestCoordinatorStreamUtil.scala

```diff
@@ -34,4 +39,31 @@ class TestCoordinatorStreamUtil {
     Mockito.verify(systemStream).getStream
     Mockito.verify(systemAdmin).createStream(any(classOf[StreamSpec]))
   }
+
+  @Test
+  def testReadConfigFromCoordinatorStream {
+    val keyForNonBlankVal = "app.id"
+    val nonBlankVal = "1"
+    val keyForEmptyVal = "task.opt"
+    val emptyVal = ""
+    val keyForNullVal = "zk.server"
+    val nullVal = null
```

Comment:
This var is unused.
In order to test the `valueAsString == null` condition, can you please add `"null"` (as a String) to the config map? That should be the case where the bytes are non-null, but the `valueAsString` is null.

Reply:
Good catch. valueSerde.toBytes(nullVal) is able to correctly encode the null value. This function "readConfigFromCoordinatorStream" is using valueSerde.fromBytes(valueAsBytes) for deserialization, so we need to pass null value to the config map (other than a string of "null").